src/tbb/CMakeLists.txt

```diff
@@ -88,6 +89,11 @@ set_target_properties(tbb PROPERTIES
     LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/def/${TBB_DEF_FILE_PREFIX}-tbb.def
 )
 
+set_property(TARGET tbb APPEND PROPERTY LINK_FLAGS_RELWITHDEBINFO ${TBB_IPO_LINK_FLAG})
+if (WIN32)
+    set_property(TARGET tbb APPEND PROPERTY LINK_FLAGS_RELWITHDEBINFO /INCREMENTAL:NO)
+endif()
+
```

Comment:
Could this property (`LINK_FLAGS_RELWITHDEBINFO`) be added inside `set_target_properties`? Is there a possibility to use [cmake-generator-expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) here to shrink this into one line?

Reply:
I tried several approaches but cannot find a way how to do it using CMake generator expressions. And also I think that the code without generator expressions looks more simple and straightforward.

Reply:
Ok, I see. Maybe it makes sense to move general part (92 line) inside `set_target_properties`.
Is it OK that we use the `-flto` flag whether we are compiling in `debug` or `relwithdebinfo`, but adding to link flags only while compiling in `relwithdebinfo`?

Reply:
@isaevil, I reworked the whole CMake part to enable such optimization for all configurations. Please review one more time.